File: 프로그래머스/연습문제/12935_제일_작은_수_제거하기.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/12935

// Todo: *min_element()

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    if (arr.size() == 1) {
        return {-1};
    }

    int minValue = *min_element(arr.begin(), arr.end());

    for (int num : arr) {
        if (num != minValue) {
            answer.push_back(num);
        }
    }

    return answer;
}

File: 프로그래머스/연습문제/12935_제일_작은_수_제거하기.java
// https://school.programmers.co.kr/learn/courses/30/lessons/12935

import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        if (arr.length == 1) {
            return new int[]{-1};
        }

        int minValue = Arrays.stream(arr).min().getAsInt();

        return Arrays.stream(arr)
                .filter(num -> num != minValue)
                .toArray();
    }
}